### PR TITLE
add removing headers on cleanup

### PR DESCRIPTION
**Pull request by Karolk99**

### CODE REVIEW

**Review by sgfn — COMMENTED** [review]
LGTM, but I don't know the plugin's inner workings enough to assess if everything is as it should be

File: lib/membrane_http_adaptive_stream/sink.ex [hunk 1]
```diff
@@ -332,13 +332,16 @@ defmodule Membrane.HTTPAdaptiveStream.Sink do
...
+        headers_to_remove = Manifest.all_headers_per_track(manifest)
+
         timeout = Membrane.Time.as_milliseconds(cleanup_after)
 
         Process.sleep(timeout)
 
         # cleanup all data of the secondary playlist and the master one
-        with {:ok, storage} <- Storage.clean_all_track_segments(storage, to_remove),
+        with {:ok, storage} <- Storage.clean_all_track_headers(storage, headers_to_remove),
+             {:ok, storage} <- Storage.clean_all_track_segments(storage, segments_to_remove),
```
Comment (mat-hek) [inline lib/membrane_http_adaptive_stream/sink.ex:344]: merge these two functions into one

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: lib/membrane_http_adaptive_stream/manifest.ex [hunk 2]
```diff
@@ -120,4 +120,9 @@ defmodule Membrane.HTTPAdaptiveStream.Manifest do
   def all_segments_per_track(%__MODULE__{} = manifest) do
     Map.new(manifest.tracks, fn {track_id, track} -> {track_id, Track.all_segments(track)} end)
   end
+
+  @spec all_header_per_track(t()) :: %{optional(track_id :: term()) => String.t()}
+  def all_header_per_track(%__MODULE__{} = manifest) do
```
Comment (mat-hek) [inline lib/membrane_http_adaptive_stream/manifest.ex:125]: what does 'all header per track' mean?

Comment (Karolk99) [reply lib/membrane_http_adaptive_stream/manifest.ex:125]: There is one header per right now. I will change the name of this function.

**Review by mat-hek — APPROVED** [review]

**Pull request merged**